scripts/bb_regression.py

```diff
@@ -0,0 +1,174 @@
+    https://github.com/skhadem/3D-BoundingBox
+"""
+
+
+import torch
+import numpy as np
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.autograd import Variable
+from top.model.backbone import resnet_fpn_backbone
+from top.run.loss_util import *
+
+
+def OrientationLoss(orient_batch, orientGT_batch, confGT_batch):
+
+    batch_size = orient_batch.size()[0]
+    indexes = torch.max(confGT_batch, dim=1)[1]
+
+    # extract just the important bin
+    orientGT_batch = orientGT_batch[torch.arange(batch_size), indexes]
+    orient_batch = orient_batch[torch.arange(batch_size), indexes]
+
+    theta_diff = torch.atan2(orientGT_batch[:,1], orientGT_batch[:,0])
+    estimated_theta_diff = torch.atan2(orient_batch[:,1], orient_batch[:,0])
+
+    return -1 * torch.cos(theta_diff - estimated_theta_diff).mean()
+
+
+class Model(nn.Module):
+    def __init__(self, features=None, bins=2, w = 0.4):
+        super(Model, self).__init__()
+        self.bins = bins
+        self.w = w
+        self.features = features
+
+        self.confidence = nn.Sequential(
+                    nn.Linear(512 * 7 * 7, 256),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(256, 256),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(256, bins),
+                    nn.Softmax()
+                )
+
+        self.translation = nn.Sequential(
+                    nn.Linear(512 * 7 * 7, 512),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(512, 512),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(512, 3)
+                )
+
+        self.orientation = nn.Sequential(
+                    nn.Linear(512 * 7 * 7, 256),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(256, 256),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(256, bins*2) # to get sin and cos
+                )
+
+        self.scale = nn.Sequential(
+                    nn.Linear(512 * 7 * 7, 512),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(512, 512),
+                    nn.ReLU(True),
+                    nn.Dropout(),
+                    nn.Linear(512, 3)
+                )
+
+    def forward(self, x):
+        x = self.features(x) # 512 x 7 x 7
+        x = x.view(-1, 512 * 7 * 7)
+        confidence = self.confidence(x)
+        translation = self.translation(x)
+        scale = self.scale(x)
+        orientation = self.orientation(x)
+        orientation = orientation.view(-1, self.bins, 2)
+        orientation = F.normalize(orientation, dim=2)
+
+        return confidence, translation, scale, orientation
+
+
+def train():
```

Comment:
I think we should seriously consider rewriting this `train()` function to utilize our training pipeline framework. If there are issues, we should resolve it there rather than solving similar problems in multiple different copies of the code for how we train these networks.